module_report/check_apis.py

```diff
@@ -0,0 +1,22 @@
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+import qtpy
+
+
+def check_apis(report_name):
+    for api in sorted(qtpy.API_NAMES):
+        # this doesn't actually change the QT_API
+        # I would prefer using this if it actually worked
+        # child_env = dict(os.environ).update({'QT_API': api})
+
+        # still required to actually change the selected API
+        os.environ['QT_API'] = api
+
+        subprocess.run(
+            [sys.executable, Path(__file__).parent / 'check_api.py', report_name],
+            check=False, # check_api.py apparently returns a non-zero exit status despite not failing
+            # env=child_env
```

Comment:
The problem was due to a silly mistake on my part—update mutates the original dict object in place and returns `None`, rather than returning a new object, so the previous code actually passed `env=None`. Using dict unpacking instead to fix this properly and even more cleanly:

```suggestion
        subprocess.run(
            [sys.executable, Path(__file__).parent / 'check_api.py', report_name],
            check=False, # check_api.py apparently returns a non-zero exit status despite not failing
            env={**os.environ, 'QT_API': api},
```

Reply:
I verified it works locally with

```console
$ python -c 'import os, subprocess; subprocess.run(["python", "-m", "qtpy", "mypy-args"])'
--always-true=PYQT5 --always-false=PYSIDE2 --always-false=PYQT6 --always-false=PYSIDE6
$ python -c 'import os, subprocess; subprocess.run(["python", "-m", "qtpy", "mypy-args"], env={**os.environ, "QT_API" : "PySide2"})'
--always-false=PYQT5 --always-true=PYSIDE2 --always-false=PYQT6 --always-false=PYSIDE6
```